### PR TITLE
[IMPROVE] New sidebar layout

### PR DESCRIPTION
**Pull request by ggazzo**
## Proposed changes

close https://github.com/RocketChat/Rocket.Chat/issues/19209

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Hotfix (a major bugfix that has to be merged asap)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by renatobecker — COMMENTED** [review]
This is my feedback after testing the new Sidedar:

- Realtime queue not working(Manual Selection algorithm):
Realtime-Queue-not-working
New chats are not appearing in the `Queue Chats` list, I need to refresh the page to see it;


- Chats remain in the `Queue Chats` list after the agent picks up the chat:
Remove-Queue-After-taking
Need to refresh the page to fix the issue


- "Mark unread" feature is no longer working, I mean, the `room name` is not highlighted anymore
Mark-Unread


- The `Hide Chat` feature is also not working anymore. The dialog appears, but when the user clicks on `Yes, hide it!`, neither the dialog is closed nor the chat is hidden.
Hide-Chat


- Omnichannel room icon is wrong
Icon
After the rewriting, the icon displayed is a `Rocket.Chat` icon instead of the traditional Omnichannel icon:  Screen Shot 2020-10-01 at 14 45 20

- Composer not available after taking the chat
Room-Read-Only-After-taking
After the agent takes the chat, the composer should be available to start chatting but, instead, a message `Room is ready only` is displayed and the agent needs to refresh the page to fix the issue.

File: app/theme/client/imports/components/sidebar/rooms-list.css [hunk 1]
```diff
@@ -44,9 +45,13 @@
...
 	&__toolbar-search {
 		position: absolute;
+
+		left: 0;
 		z-index: 10;
 
-		left: 10px;
+		padding-block-start: 12px;
+
+		height: 100vh;
```
Comment (tassoevan) [inline app/theme/client/imports/components/sidebar/rooms-list.css:54]: Does it not fail when the banner is shown?

File: app/ui-sidenav/client/sideNav.js [hunk 2]
```diff
@@ -8,6 +9,10 @@ import { SideNav, menu } from '../../ui-utils';
 import { settings } from '../../settings';
 import { roomTypes, getUserPreference } from '../../utils';
 import { Users } from '../../models';
+import { createTemplateForComponent } from '../../../client/reactAdapters';
+
+createTemplateForComponent('sidebarHeader', () => import('../../../client/components/sidebar/Header'));
+createTemplateForComponent('sidebarChats', () => import('../../../client/components/sidebar/Chats'), { renderContainerView: () => HTML.DIV({ style: 'display: flex; flex: 1 1 auto;' }) });// eslint-disable-line new-cap
```
Comment (tassoevan) [inline app/ui-sidenav/client/sideNav.js:15]: Why this is style is needed on the template's wrapper `div`? Could it be inside the component itself?

File: app/ui-sidenav/client/sidebarHeader.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<template name="sidebarHeader">
+<template name="sidebarHeader2">
```
Comment (tassoevan) [inline app/ui-sidenav/client/sidebarHeader.html:1]: Why?

**Review by MartinSchoeler — APPROVED** [review]

**Pull request merged**